uni/lib/view/common_widgets/pages_layouts/secondary/secondary.dart

```diff
@@ -1,21 +1,42 @@
+import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 import 'package:uni/view/common_widgets/pages_layouts/general/general.dart';
-import 'package:uni/view/common_widgets/pages_layouts/general/widgets/app_bar.dart';
+import 'package:uni/view/common_widgets/pages_layouts/general/widgets/bottom_navigation_bar.dart';
 import 'package:uni/view/common_widgets/pages_layouts/general/widgets/refresh_state.dart';
+import 'package:uni/view/common_widgets/pages_layouts/general/widgets/top_navigation_bar.dart';
 
 /// Page with a back button on top
 abstract class SecondaryPageViewState<T extends StatefulWidget>
     extends GeneralPageViewState<T> {
   @override
   Scaffold getScaffold(BuildContext context, Widget body) {
     return Scaffold(
-      appBar: CustomAppBar(getTopRightButton: getTopRightButton),
+      appBar: getTopNavbar(context),
+      bottomNavigationBar: AppBottomNavbar(
+        parentContext: context,
+      ),
       body: RefreshState(onRefresh: onRefresh, child: body),
     );
   }
 
+  String? getTitle() {
+    return null;
+  }
```

Comment:
Given that we (usually) want these pages to show a title, I would remove the default body here and force implementations to define this.